homeassistant/components/vacuum/neato.py

```diff
@@ -0,0 +1,215 @@
+"""
+Support for Neato Connected Vaccums.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/vacuum.neato/
+"""
+import logging
+
+import requests
+
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.components.vacuum import (
+  VacuumDevice, SUPPORT_BATTERY, SUPPORT_PAUSE, SUPPORT_RETURN_HOME,
+  SUPPORT_STATUS, SUPPORT_STOP, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+  SUPPORT_MAP, ATTR_STATUS, ATTR_BATTERY_LEVEL, ATTR_BATTERY_ICON)
+from homeassistant.components.neato import (
+    NEATO_ROBOTS, NEATO_LOGIN, NEATO_MAP_DATA, ACTION, ERRORS, MODE, ALERTS)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['neato']
+
+SUPPORT_NEATO = SUPPORT_BATTERY | SUPPORT_PAUSE | SUPPORT_RETURN_HOME | \
+                 SUPPORT_STOP | SUPPORT_TURN_OFF | SUPPORT_TURN_ON | \
+                 SUPPORT_STATUS | SUPPORT_MAP
+
+ICON = "mdi:roomba"
+
+ATTR_CLEAN_START = 'clean_start'
+ATTR_CLEAN_STOP = 'clean_stop'
+ATTR_CLEAN_AREA = 'clean_area'
+ATTR_CLEAN_BATTERY_START = 'battery_level_at_clean_start'
+ATTR_CLEAN_BATTERY_END = 'battery_level_at_clean_end'
+ATTR_CLEAN_SUSP_COUNT = 'clean_suspension_count'
+ATTR_CLEAN_SUSP_TIME = 'clean_suspension_time'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Neato vacuum."""
+    dev = []
+    for robot in hass.data[NEATO_ROBOTS]:
+        dev.append(NeatoConnectedVacuum(hass, robot))
+    _LOGGER.debug("Adding vacuums %s", dev)
+    add_devices(dev)
+
+
+class NeatoConnectedVacuum(VacuumDevice):
+    """Neato Connected Vacuums."""
+
+    def __init__(self, hass, robot):
+        """Initialize the Neato Connected Vacuums."""
+        self.robot = robot
+        self.neato = hass.data[NEATO_LOGIN]
+        self._name = '{}'.format(self.robot.name)
+        self._status_state = None
+        self._clean_state = None
+        self._state = None
+        self._mapdata = hass.data[NEATO_MAP_DATA]
+        self.clean_time_start = None
+        self.clean_time_stop = None
+        self.clean_area = None
+        self.clean_battery_start = None
+        self.clean_battery_end = None
+        self.clean_suspension_charge_count = None
+        self.clean_suspension_time = None
+        self.update()
```

Comment:
Remove this and instead call `add_devices` in `setup_platform` like this:
```py
add_devices(dev, True)
```
`update` will then be called during adding the devices to home assistant.